Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/RayCastLidar.cpp

```diff
@@ -150,16 +150,34 @@ ARayCastLidar::FDetection ARayCastLidar::ComputeDetection(const FHitResult& HitI
       PointsPerChannel[idxChannel] = RecordedHits[idxChannel].size();
 
     LidarData.ResetMemory(PointsPerChannel);
+    PointCloudResetMemory();
 
     for (auto idxChannel = 0u; idxChannel < Description.Channels; ++idxChannel) {
       for (auto& hit : RecordedHits[idxChannel]) {
         FDetection Detection = ComputeDetection(hit, SensorTransform);
         if (PostprocessDetection(Detection))
+        {
           LidarData.WritePointSync(Detection);
+          PointCloudWritePointSync(Detection);
```

Comment:
Would be nice if this operations is surrounded with an if condition and #if WITH_EDITOR

Reply:
Fixed.